src/main/kotlin/Task2.3.kt

```diff
@@ -0,0 +1,72 @@
+fun main() {
+    val arrayOfNumbers = Array(10) { _ -> (1..20).random() }
+
+    println("Random string:")
+    arrayOfNumbers.forEach { i -> print("$i ") }
+    println(arrayOfNumbers.forEach { i -> i+1 })
+    println("")
+    println("")
+
+    var result = getMulAndExtremesByFor(arrayOfNumbers)
+    println("Results by getMulAndExtremesByFor: mul is ${result[0]}, min is ${result[1]}, max is ${result[2]}")
+
+    result = getMulAndExtremesByWhile(arrayOfNumbers)
+    println("Results by getMulAndExtremesByWhile: mul is ${result[0]}, min is ${result[1]}, max is ${result[2]}")
+
+    result = getMulAndExtremesByReduce(arrayOfNumbers)
+    println("Results by getMulAndExtremesByReduce: mul is ${result[0]}, min is ${result[1]}, max is ${result[2]}")
+
+    result = getMulAndExtremesByForEach(arrayOfNumbers)
+    println("Results by getMulAndExtremesByForEach: mul is ${result[0]}, min is ${result[1]}, max is ${result[2]}")
+
+    println("Results by min and max: min is ${arrayOfNumbers.minOrNull()}, max is ${arrayOfNumbers.maxOrNull()}")
+}
+
+fun getMulAndExtremesByFor(arr: Array<Int>): Array<Int> {
+    var mul = 1
+    var max = arr.first()
+    var min = arr.first()
+
+    for (value in arr){
+        mul *= value
+        if (max < value) max = value
+        if (min > value) min = value
+    }
+
+    return arrayOf(mul, min, max)
+}
+
+fun getMulAndExtremesByWhile(arr: Array<Int>): Array<Int> {
+    var mul = 1
+    var max = arr.first()
+    var min = arr.first()
+
+    var i = 0
+    while (i < arr.size) {
+        mul *= arr[i]
+        if (max < arr[i]) max = arr[i]
+        if (min > arr[i]) min = arr[i]
+        i++
+    }
+
+    return arrayOf(mul, min, max)
+}
+
+fun getMulAndExtremesByReduce(arr: Array<Int>): Array<Int> {
+    val mul = arr.reduce{ innerMul, elem -> innerMul * elem }
```

Comment:
На будущее старайся после именованных параметров и -> выполнение переносить на новую строку
```suggestion
    val mul = arr.reduce{ innerMul, elem ->
     innerMul * elem 
     }
```

Reply:
Принял